websites/T/Tubi/presence.ts

```diff
@@ -1,41 +1,101 @@
+		{
+			play: "general.playing",
+			paused: "general.paused",
+			browse: "general.browsing",
+			live: "general.live",
+		},
+		await presence.getSetting<string>("lang").catch(() => "en")
+	);
+}
+enum Assets {
+	Logo = "https://i.imgur.com/PfRmgZm.png",
+	Live = "https://i.imgur.com/gYNKPnn.png",
+	Play = "https://i.imgur.com/OLaz6JN.png",
+	Paused = "https://i.imgur.com/4iyMINk.png",
+	SearchImage = "https://i.imgur.com/oGQtnIY.png",
+}
+let strings: Awaited<ReturnType<typeof getStrings>>,
+	oldLang: string = null;
 
 presence.on("UpdateData", async () => {
 	const presenceData: PresenceData = {
-			largeImageKey: "tubi-logo",
+			largeImageKey: Assets.Logo,
 		},
-		video: HTMLVideoElement = document.querySelector(
-			"video#videoPlayerComponent"
-		);
-	if (video && !isNaN(video.duration)) {
-		const [startTimestamp, endTimestamp] = presence.getTimestamps(
-			Math.floor(video.currentTime),
-			Math.floor(video.duration)
-		);
-		presenceData.details = document
-			.querySelector('meta[property="og:title"]')
-			.getAttribute("content");
-		presenceData.smallImageKey = video.paused ? "pause" : "play";
-		presenceData.smallImageText = video.paused
-			? (await strings).pause
-			: (await strings).play;
-		presenceData.startTimestamp = startTimestamp;
-		presenceData.endTimestamp = endTimestamp;
+		video = document.querySelector<HTMLVideoElement>("video"),
+		search = document.querySelector<HTMLInputElement>('[type="search"]'),
+		{ href, pathname } = document.location,
+		[newLang, privacy, buttons, covers] = await Promise.all([
+			presence.getSetting<string>("lang").catch(() => "en"),
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("buttons"),
+			presence.getSetting<boolean>("covers"),
+		]);
+	if (oldLang !== newLang || !strings) {
+		oldLang = newLang;
+		strings = await getStrings();
+	}
+
+	if (privacy) {
+		presenceData.details = strings.browse;
+		return;
+	}
+	if (search?.value) {
+		presenceData.details = "Searching for";
+		presenceData.state = search.value;
+		presenceData.smallImageKey = Assets.SearchImage;
+	} else if (pathname.includes("/category/")) {
+		presenceData.details = "Viewing category";
+		presenceData.state = document.querySelector<HTMLMetaElement>(
+			'[property="og:title"]'
+		).content;
+	} else if (video && !isNaN(video.duration)) {
+		presenceData.largeImageKey = document.querySelector<HTMLMetaElement>(
+			'meta[property="og:image"]'
+		).content;
+		presenceData.details = document.querySelector<HTMLMetaElement>(
+			'meta[property="og:title"]'
+		).content;
+		if (!pathname.includes("live")) {
+			presenceData.smallImageKey = video.paused ? Assets.Paused : Assets.Play;
+			presenceData.smallImageText = video.paused
+				? strings.paused
+				: strings.play;
+			presenceData.endTimestamp = presence.getTimestamps(
+				Math.floor(video.currentTime),
+				Math.floor(video.duration)
+			)[1];
+			presenceData.buttons = [
+				{
+					label: "Watch Video",
```

Comment:
No string for that?

Reply:
Exist in general strings :
https://github.com/PreMiD/Localization/blob/main/src/Presence/general.json#L318-L320